### ISSUE TITLE
Rails 4 error - undefined method `graft' when setting up FFCRM as an Engine

### ISSUE THREAD

**Issue by colakatz** [fatfreecrm/fat_free_crm#388]
While trying to install FFCRM as an engine, I came across the following error during setup:

NameError: undefined method `graft' for class `ActiveRecord::Associations::JoinDependency'

This specific error comes up when I tried running the command "rake db:migrate ffcrm:setup:admin" as I setup the engine.  However, a similar error message comes up when I just tried running the built in web server via "rails server", specifically:

c:/RailsInstaller/Ruby2.0.0/lib/ruby/gems/2.0.0/gems/activesupport-4.1.0/lib/active_support/core_ext/module/aliasing.rb:32:in `alias_method': undefined method `graft' for class `ActiveRecord::Associations::JoinDependency' (NameError)

My research online that I did found that the graft function was removed from Rails 4.1 and that it is used as part of the Polyamorous gem which is a dependency of the ransack gem.  It appears that the version used by ransack 1.0.0 (the version defined in the FFCRM gemspec file uses a version of Polyamorous where this is not fixed, 0.6.4).  I gathered this information mostly from an internet search of the above error and from this issue... https://github.com/activerecord-hackery/polyamorous/issues/6

**Comment by steveyken** [comment]
Thanks for reporting and tracking down. Good details to know. Are you using the rails4 branch of FFCRM or master? Master isn't ready for Rails 4 yet, but we are actively moving towards that point. It still needs engine testing though which is next on my list.

**Comment by colakatz** [comment]
Hi Steve,

I am using the Rails 4 branch for this.

**Comment by colakatz** [comment]
I should also mention that in addition to using the Rails4 branch, you
cannot run the engine built in one version of Rails (at least major
versions) in an app that is built in another version.  I tried that when I
first started looking at FFCRM and when I tried using the master branch in
a Rails 4.1 app, it gave me a number of errors.  Rather than deal with it,
I decided to use the master and made my changes to a local copy of that.

However, I like the idea of the engine version as it gives me a chance to
do what I need to do and makes it simple to keep it up to date with the bug
fixes.

**Comment by warp** [comment]
The rails 4 branch has several gem dependencies currently pointing to git repos to work around this issue you will need to include those same dependencies in your application Gemfile
```
gem 'rails3-jquery-autocomplete', github: 'crowdint/rails3-jquery-autocomplete'
gem 'ransack', github: 'activerecord-hackery/ransack'
gem 'ransack_ui', github: 'fatfreecrm/ransack_ui', branch: 'fix/ransortable_attributes'
```

**Comment by melissawahnish** [comment]
Hi!  I am also working from the rails4 branch and the above didn't work for me.  I received this message when I ran bundle:

Could not find gem 'ransack (= 1.1.0) ruby', which is required by gem
'fat_free_crm (>= 0) ruby', in any of the sources.

I then pulled down the gem and modified fat_free_crm.gemspec to depend on ransack 1.5.1. Using my local version of the gem,  bundle then completed fine.

Hope that helps!
